File: ruby/testcode/securecookie_004.rb
require_relative 'shared'

# vuln-code-snippet start ruby_securecookie_rack_secure
def set_session(req)
  session_id = req.param('session_id')
  cookie_opts = { value: session_id, secure: true, httponly: true, same_site: 'Strict', path: '/' }
  header = Rack::Utils.add_cookie_to_header(nil, 'session', cookie_opts)  # vuln-code-snippet safe-line ruby_securecookie_rack_secure
  response = BenchmarkResponse.ok('session started')
  response.headers['Set-Cookie'] = header
  response
end
# vuln-code-snippet end ruby_securecookie_rack_secure

File: rust/testcode/securecookie_004.rs
//! CWE-614: Cookie set with SameSite=None but missing required Secure attribute.

// vuln-code-snippet start testcodeSecurecookie004
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let token = req.param("token");

    let cookie = format!("session={}; Path=/; SameSite=None", token); // vuln-code-snippet target-line testcodeSecurecookie004

    super::shared::BenchmarkResponse::ok(&format!("Set-Cookie: {}", cookie))
}
// vuln-code-snippet end testcodeSecurecookie004
